### PR TITLE
Moved user input to the top of the scripts.

### LINKED ISSUE #108: Move user input to the top of every script.
Every user input should be at the start of scripts.

### PR DESCRIPTION
**Pull request by ludeeus**
### Description:
 - Moved all user input to the top of the scripts.
  _Cosmetic change, so the user can go get a cup of coffee/tea while the script run, and not worry about it hanging waiting for input._
 - Where there was an validation this line is not needed an has been removed:
"If you have issues with this script, please say something in the #devs_hassbian channel on Discord."
_This line will still print if the validation fail._
- Added validation to appdaemon-install-package

**Related issue (if applicable):** Fixes #108 

### Checklist:
  - [X] The code change is tested and works locally.
  - [ ] Script has validation check of the job.
    - Not really applicable here. 
  
#### If pertinent:
  - [ ] Created/Updated documentation at `/docs`
    - Not really applicable here.

### CODE REVIEW

File: package/opt/hassbian/suites/appdaemon.sh [hunk 1]
```diff
@@ -15,6 +15,13 @@ function appdaemon-install-package {
 appdaemon-show-short-info
 appdaemon-show-copyright-info
 
+if [ "$ACCEPT" != "true" ]; then
+  if [ -f "/usr/sbin/samba" ]; then
+    echo -n "Do you want to add samba share for AppDaemon configuration? [N/y] : "
```
Comment (Landrash) [inline package/opt/hassbian/suites/appdaemon.sh]: Samba with capital S

File: package/opt/hassbian/suites/appdaemon.sh [hunk 2]
```diff
@@ -53,34 +60,37 @@ sync
...
-echo "To continue have a look at http://appdaemon.readthedocs.io/en/latest/"
-echo
-echo "If you have issues with this script, please say something in the #devs_hassbian channel on Discord."
-echo
+validation=$(ps -ef | grep -v grep | grep appdaemon | wc -l)
+if [ "$validation" != "0" ]; then
+	echo
+	echo -e "\e[32mInstallation done..\e[0m"
+	echo
+  echo "You will find the appdaemon configuration files in:"
```
Comment (Landrash) [inline package/opt/hassbian/suites/appdaemon.sh]: Formatting white space missing here and two line more.

Comment (Landrash) [inline package/opt/hassbian/suites/appdaemon.sh]: Appdaemon with Capital A?

File: package/opt/hassbian/suites/homebridge.sh [hunk 3]
```diff
@@ -17,43 +17,49 @@ function homebridge-install-package {
...
+  if [ -f "/usr/sbin/samba" ]; then
+    echo -n "Do you want to add samba share for Homebridge configuration? [N/y] : "
+    read SAMBA
+  fi
+fi
+
+echo "Preparing system, and adding dependencies..."
+sudo apt update
+sudo apt -y upgrade
+sudo apt install -y make git
```
Comment (Landrash) [inline package/opt/hassbian/suites/homebridge.sh]: Isn't `git` and `make` both installed in the base image?

File: package/opt/hassbian/suites/homebridge.sh [hunk 4]
```diff
@@ -17,43 +17,49 @@ function homebridge-install-package {
...
   echo ""
   echo ""
   echo -n "Enter your Home Assistant API password: "
   read -s HOMEASSISTANT_PASSWORD
   echo
 fi
 
+if [ "$ACCEPT" != "true" ]; then
+  if [ -f "/usr/sbin/samba" ]; then
+    echo -n "Do you want to add samba share for Homebridge configuration? [N/y] : "
```
Comment (Landrash) [inline package/opt/hassbian/suites/homebridge.sh]: Capital S for Samba.

File: package/opt/hassbian/suites/homebridge.sh [hunk 5]
```diff
@@ -17,43 +17,49 @@ function homebridge-install-package {
...
+
+echo "Preparing system, and adding dependencies..."
+sudo apt update
+sudo apt -y upgrade
+sudo apt install -y make git
+curl -sL https://deb.nodesource.com/setup_8.x | sudo -E bash -
+sudo apt install -y nodejs
+sudo apt install -y libavahi-compat-libdnssd-dev
+
+echo "Installing homebridge for homeassistant..."
```
Comment (Landrash) [inline package/opt/hassbian/suites/homebridge.sh]: Capital H for Homebridge, captial for Home Assistant and spelling for Home Assistant.

File: package/opt/hassbian/suites/homebridge.sh [hunk 6]
```diff
@@ -108,24 +114,19 @@ sudo systemctl daemon-reload
...
-			echo "create mask = 0644" | tee -a /etc/samba/smb.conf
-			echo "directory mask = 0755" | tee -a /etc/samba/smb.conf
-			echo "force user = homebridge" | tee -a /etc/samba/smb.conf
-			echo "" | tee -a /etc/samba/smb.conf
-			echo "Restarting Samba service"
-			sudo systemctl restart smbd.service
-		fi
-	fi
+if [ "$SAMBA" == "y" ] || [ "$SAMBA" == "Y" ]; then
+	echo "Adding configuration to samba..."
```
Comment (Landrash) [inline package/opt/hassbian/suites/homebridge.sh]: Capital S in Samba

File: package/opt/hassbian/suites/homebridge.sh [hunk 7]
```diff
@@ -139,9 +140,7 @@ if [ "$validation" != "0" ]; then
 	echo "use this: '$HOMEBRIDGE_PIN'"
 	echo "For more information see this repo:"
 	echo "https://github.com/home-assistant/homebridge-homeassistant#customization"
-	echo
-	echo "If you have issues with this script, please say something in the #devs_hassbian channel on Discord."
-	echo
+  echo
```
Comment (Landrash) [inline package/opt/hassbian/suites/homebridge.sh]: Formatting white space missing.

**Review by Landrash — CHANGES_REQUESTED** [review]
Some spelling and formatting will need to be looked at but looks good in general.

File: package/opt/hassbian/suites/appdaemon.sh (hunk 1 above)
Comment (ludeeus) [reply package/opt/hassbian/suites/appdaemon.sh]: 👍

File: package/opt/hassbian/suites/appdaemon.sh (hunk 2 above)
Comment (ludeeus) [reply package/opt/hassbian/suites/appdaemon.sh]: Not sure how :/

Comment (ludeeus) [reply package/opt/hassbian/suites/appdaemon.sh]: Capital D to :)

File: package/opt/hassbian/suites/homebridge.sh (hunk 4 above)
Comment (ludeeus) [reply package/opt/hassbian/suites/homebridge.sh]: 👍

File: package/opt/hassbian/suites/appdaemon.sh (hunk 2 above)
Comment (Landrash) [reply package/opt/hassbian/suites/appdaemon.sh]: It's a "TAB" that's missing before the `echo`

Comment (ludeeus) [reply package/opt/hassbian/suites/appdaemon.sh]: Yeah, I see that is missing, not sure how that happened :)

File: package/opt/hassbian/suites/homebridge.sh (hunk 3 above)
Comment (ludeeus) [reply package/opt/hassbian/suites/homebridge.sh]: They sure are :)

File: package/opt/hassbian/suites/homebridge.sh (hunk 5 above)
Comment (ludeeus) [reply package/opt/hassbian/suites/homebridge.sh]: 👍

File: package/opt/hassbian/suites/homebridge.sh (hunk 6 above)
Comment (ludeeus) [reply package/opt/hassbian/suites/homebridge.sh]: :+1:

File: package/opt/hassbian/suites/homebridge.sh (hunk 7 above)
Comment (ludeeus) [reply package/opt/hassbian/suites/homebridge.sh]: :+1:

**Comment by ludeeus** [comment]
Found some more spelling errors :)

File: package/opt/hassbian/suites/appdaemon.sh [hunk 8]
```diff
@@ -80,7 +80,7 @@ if [ "$validation" != "0" ]; then
 	echo
 	echo -e "\e[32mInstallation done..\e[0m"
 	echo
-  echo "You will find the appdaemon configuration files in:"
+	echo "You will find the AppDaemon configuration files in:"
   echo "/home/homeassistant/appdaemon"
```
Comment (Landrash) [inline package/opt/hassbian/suites/appdaemon.sh]: Missing white space.

Comment (Landrash) [inline package/opt/hassbian/suites/appdaemon.sh]: Missing white space.

**Review by Landrash — CHANGES_REQUESTED** [review]

**Review by Landrash — APPROVED** [review]
Looks good!
Great work as usual :cake:

**Pull request merged**